### PR TITLE
add tests and api orders status

### PR DESCRIPTION
**Pull request by ClaireAlma**
### Reason for change

[ClickUp task](https://linear.app/almapay/issue/ECOM-1667/[%F0%9F%A7%A9-php-client]-implement-order-status-endpoint)

### How to test
Launch the unit tests

### CODE REVIEW

**Review by Francois-Gomis — APPROVED** [review]

**Review by Francois-Gomis — APPROVED** [review]

**Review by joyet-simon — APPROVED** [review]
Tests done

File: src/Endpoints/Orders.php [hunk 1]
```diff
@@ -90,4 +108,57 @@ public function fetch($orderId)
...
+     * @throws RequestException
+     */
+    public function sendStatus($orderExternalId, $orderData = array())
+    {
+        $this->validateStatusData($orderData);
+        $label = $this->arrayUtils->slugify($orderData['label']);
+
+        try {
+            $response = $this->request(self::ORDERS_PATH_V2 . "/{$orderExternalId}/status")->setRequestBody(array(
+                'label' =>  $label,
```
Comment (Francois-Gomis) [inline src/Endpoints/Orders.php:126]: can you change label by status . Sorry

File: src/Endpoints/Orders.php [hunk 2]
```diff
@@ -90,4 +108,57 @@ public function fetch($orderId)
...
+        } catch (MissingKeyException $e ) {
+            throw new ParametersException('Error in the required parameters (label, is_shipped) when calling orders.sendStatus',null,  $e);
+        }
+
+        if(!is_bool($orderData['is_shipped'])) {
+            throw new ParametersException('Parameter "is_shipped" must be a boolean');
+        }
+
+        if(!$orderData['label']) {
+            throw new ParametersException('Missing the required parameter "label" when calling orders.sendStatus');
```
Comment (Francois-Gomis) [inline src/Endpoints/Orders.php:161]: status

File: src/Lib/ArrayUtils.php [hunk 3]
```diff
@@ -55,8 +57,57 @@ public function checkMandatoryKeys($keys, $array)
...
+     *
+     * @param string $textToSlugify
+     *
+     * @throws ParametersException
+     *
+     * @return String
+     */
+    public function slugify($textToSlugify){
+        if(!is_string($textToSlugify)) {
+            throw new ParametersException('Label must be a string');
```
Comment (Francois-Gomis) [inline src/Lib/ArrayUtils.php:81]: status

File: src/Lib/ArrayUtils.php [hunk 4]
```diff
@@ -55,8 +57,57 @@ public function checkMandatoryKeys($keys, $array)
...
+        $text = preg_replace('~-+~', '-', $text);
+
+        // lowercase
+        $text = strtolower($text);
+
+        if (
+            empty($text)
+            || '_' === $text
+        ) {
+            throw new ParametersException(sprintf('The slug is empty, label "%s"', $textToSlugify));
```
Comment (Francois-Gomis) [inline src/Lib/ArrayUtils.php:107]: status

File: tests/Unit/Legacy/Endpoints/OrdersTest.php [hunk 5]
```diff
@@ -0,0 +1,162 @@
...
+        $this->orderEndpoint = \Mockery::mock(Orders::class)->makePartial();
+        $this->arrayUtils = \Mockery::mock(ArrayUtils::class)->makePartial();
+        $this->responseMock = \Mockery::mock(Response::class);
+        $this->requestObject = \Mockery::mock(Request::class);
+        $loggerMock = \Mockery::mock(LoggerInterface::class);
+        $loggerMock->shouldReceive('error');
+
+        $this->orderEndpoint->arrayUtils = $this->arrayUtils;
+        $this->externalId  = 'Mon external Id';
+        $this->label = 'Mon Label - 1';
```
Comment (Francois-Gomis) [inline tests/Unit/Legacy/Endpoints/OrdersTest.php:38]: status in this file

File: tests/Unit/Legacy/Lib/ArrayUtilsTest.php [hunk 6]
```diff
@@ -0,0 +1,43 @@
...
+
+    public function testSlugifyNotString()
+    {
+        $this->expectException(ParametersException::class);
+        $this->arrayUtils->slugify(array());
+    }
+
+    public function testSlugifyOk()
+    {
+        $this->assertEquals('mon_label_1', $this->arrayUtils->slugify('Mon label 1'));
```
Comment (Francois-Gomis) [inline tests/Unit/Legacy/Lib/ArrayUtilsTest.php:25]: Status in this file

File: src/Endpoints/Orders.php (hunk 2 above)
Comment (ClaireAlma) [reply src/Endpoints/Orders.php]: Done

File: tests/Unit/Legacy/Endpoints/OrdersTest.php (hunk 5 above)
Comment (ClaireAlma) [reply tests/Unit/Legacy/Endpoints/OrdersTest.php]: Done

File: src/Endpoints/Orders.php (hunk 1 above)
Comment (ClaireAlma) [reply src/Endpoints/Orders.php]: Done

**Review by joyet-simon — APPROVED** [review]

File: src/Endpoints/Orders.php [hunk 7]
```diff
@@ -26,12 +26,30 @@
...
 
 class Orders extends Base
 {
+    /**
+     * @var ArrayUtils
+     */
+    public $arrayUtils;
+
     const ORDERS_PATH = '/v1/orders';
+    const ORDERS_PATH_V2 = '/2/orders';
```
[the remark this answers is not in the document]
Comment (ClaireAlma) [reply src/Endpoints/Orders.php]: Done

**Pull request merged**